### PR TITLE
Dev test ifexpr

### PR DESCRIPTION
**Pull request by RyanDJLee**
- Update documentation for IfExpr node.
- Implement IfExpr node visitor.
- Add test cases for IfExpr node representing an assignment of an if expression to a variable with both body/orelse suites having the same type as well as different types.

### CODE REVIEW

**Comment by RyanDJLee** [comment]
- Will update using hypothesis testing.

**Comment by RyanDJLee** [comment]
- Build is breaking due to name change in custom_hypothesis_support; fixed issue in separate branch.

**Comment by david-yz-liu** [comment]
@RyanDJLee you should fix the error in this branch.

File: python_ta/transforms/type_inference_visitor.py [hunk 1]
```diff
@@ -302,6 +302,12 @@ def visit_for(self, node):
             for i in range(len(target_tvars)):
                 self.type_constraints.unify(rtype.__args__[0], target_tvars[i])
 
+    def visit_ifexp(self, node):
+        if node.body.type_constraints.type == node.orelse.type_constraints.type:
```
Comment (david-yz-liu) [inline python_ta/transforms/type_inference_visitor.py]: There should be a `unify` here.

File: tests/custom_hypothesis_support.py [hunk 2]
```diff
@@ -36,6 +36,14 @@
 numeric_values = numeric_types.flatmap(lambda s: s())
 
 
+eval_types = hs.sampled_from([
```
Comment (david-yz-liu) [inline tests/custom_hypothesis_support.py]: Another strategy is probably not necessary here.

**Pull request merged**